# What is money?

## Questions

1. Why is simply counting currency an inadequate measure of money?

    Checks, checking account deposits and wealth also count as money.

2. In prison, cigarettes are sometimes used among inmates as a form of payment. How is it possible for cigarettes to solve the “double coincidence of wants” problem, even if a prisoner does not smoke?

    As long as cigarettes can exchange anything, have storage value and as an unit of account, they become money.

3. Three goods are produced in an economy by three individuals:

| *Good* | *Producer* |
| --- | ---: |
| Apples | Orchard owner |
| Bananas | Banana grower |
| Chocolate | Chocolatier |

    If the orchard owner likes only bananas, the banana grower likes only chocolate, and the chocolatier likes only apples, will any trade between these three persons take place in a barter economy? How will introducing money into the economy benefit these three producers?
    
    No. They don't need to waste time on finding "double coincidence of wants".
    
4. Why did cavemen not need money?

    They don't have much things to exchange.
    
5. Most of the time it is quite difficult to separate the three functions of money. Money performs its three functions at all times, but sometimes we can stress one in particular. For each of the following situations, identify which function of money is emphasized.

    a. Brooke accepts money in exchange for performing her daily tasks at her office, since she knows she can use that money to buy goods and services.
    
    b. Tim wants to calculate the relative value of oranges and apples, and therefore checks the price per pound of each of these goods as quoted in currency units.
    
    c. Maria is currently pregnant. She expects her expenditures to increase in the future and decides to increase the balance in her savings account.
    
    a: medium of exchange
    
    b: unit of account
    
    c: store of value
    
6. In Brazil, a country that underwent a rapid inflation before 1994, many transactions were conducted in dollars rather than in reals, the domestic currency. Why?

    Inflation made Brazilians didn't trust their own currency.
    
7. Was money a better store of value in the United States in the 1950s than in the 1970s? Why or why not? In which period would you have been more willing to hold money?

    ![](https://fred.stlouisfed.org/graph/fredgraph.png?g=lXs0)
    
    Inflation is higher in the 1970s, so it's better hold money in the 1950s.
    
8. Why have some economists described money during a hyperinflation as a “hot potato” that is quickly passed from one person to another?

    You don't want hold bunch of waste paper.
    
9. Why were people in the United States in the nineteenth century sometimes willing to be paid by check rather than with gold, even though they knew there was a possibility that the check might bounce?

    Gold is heavy.
    
10. In ancient Greece, why was gold a more likely candidate for use as money than wine?

    Wine doesn't meet most of the criteria of money:
    
    (1) It must be easily standardized, making it simple to ascertain its value;
    
    (2) it must be widely accepted;
    
    (3) it must be divisible, so that it is easy to “make change”;
    
    (4) it must be easy to carry;
    
    (5) it must not deteriorate quickly.
    
11. If you use an online payment system such as PayPal to purchase goods or services on the Internet, does this affect the M1 money supply, the M2 money supply, both, or neither? Explain.

    PayPal needs to link to debit card or credit card, that will affect demand deposit, so it also affect both M1 and M2.
    
12. Rank the following assets from most liquid to least liquid:

    a. Checking account deposits
    
    b. Houses
    
    c. Currency
    
    d. Automobiles
    
    e. Savings deposits
    
    f. Common stock
    
    c a e f d b
    
13. Which of the Federal Reserve’s measures of the monetary aggregates—M1 or M2—is composed of the most liquid assets? Which is the larger measure?

    M1, M2.
    
14. It is not unusual to find a business that displays a sign saying “no personal checks, please.” On the basis of this observation, comment on the relative degree of liquidity of a checking account versus currency.

    Checking account is less liquid then currency.
    
15. For each of the following assets, indicate which of the monetary aggregates (M1 and M2) includes them:

    a. Currency
    
    b. Money market mutual funds
    
    c. Small-denomination time deposits
    
    d. Checkable deposits
    
    M1: a d
    
    M2: b c
    
16. Assume that you are interested in earning some return on the idle balances you usually keep in your checking account and decide to buy some money market mutual funds shares by writing a check. Comment on the effect of your action (with everything else the same) on M1 and M2.

    M1 will decrese, M2 doesn't change.
    
17. In April 2009, the growth rate of M1 fell to 6.1%, while the growth rate of M2 rose to 10.3%. In September 2013, the year-over-year growth rate of the M1 money supply was 6.5%, while the growth rate of the M2 money supply was about 8.3%. How should Federal Reserve policymakers interpret these changes in the growth rates of M1 and M2?

    From 2009 to 2013, the M1 growth rate increased by 0.4%, the growth rate of M2 fell by 2%. They moved in the opposite direction, it's hard for policymakers to choose the best policy.
    
18. Suppose a researcher discovers that a measure of the total amount of debt in the U.S. economy over the past twenty years was a better predictor of inflation and the business cycle than M1 or M2. Does this discovery mean that we should define money as equal to the total amount of debt in the economy?

    If there are no debts, there will be no money. That doesn't make sense.
    
## Applied problems

19. The table below shows hypothetical values, in billions of dollars, of different forms of money.

    a. Use the table to calculate the M1 and M2 money supplies for each year, as well as the growth rates of the M1 and M2 money supplies from the previous year.
    
    b. Why are the growth rates of M1 and M2 so different? Explain.

In [2]:
import pandas as pd
indexes = ["Currency", "Money market mutual fund shares", "Saving account deposits",
           "Money market deposit accounts", "Demand and checkable deposits", "Small-denomination time deposits", 
           "Traveler’s checks", "3-month Treasury bills"]
columns = [2009, 2010, 2011, 2012]
data = [[950,     960,   962,   967],
        [677,     678,   676,   685],
        [5_600, 5_880, 6_068, 6_205],
        [1_200, 1_231, 1_260, 1_315],
        [1_010,   982,   990, 1_003],
        [  830,   861, 1_123, 1_566],
        [    4,     4,     3,     2],
        [1_996, 2_384, 2_446, 2_512]]
df = pd.DataFrame(data, index=indexes, columns=columns)
df

2009  2010  2011  2012
Currency                           950   960   962   967
Money market mutual fund shares    677   678   676   685
Saving account deposits           5600  5880  6068  6205
Money market deposit accounts     1200  1231  1260  1315
Demand and checkable deposits     1010   982   990  1003
Small-denomination time deposits   830   861  1123  1566
Traveler’s checks                    4     4     3     2
3-month Treasury bills            1996  2384  2446  2512

In [39]:
M1_assets = ["Currency", "Traveler’s checks", "Demand and checkable deposits"]
M1_df = df.loc[M1_assets]
M1_df

2009  2010  2011  2012
Currency                        950   960   962   967
Traveler’s checks                 4     4     3     2
Demand and checkable deposits  1010   982   990  1003

In [40]:
M1_df.sum()

2009    1964
2010    1946
2011    1955
2012    1972
dtype: int64

In [52]:
# M1 growth rate
M1_df.sum().pct_change()

2009         NaN
2010   -0.009165
2011    0.004625
2012    0.008696
dtype: float64

In [41]:
M2_df = df.loc[~df.index.isin(["3-month Treasury bills"])]
M2_df

2009  2010  2011  2012
Currency                           950   960   962   967
Money market mutual fund shares    677   678   676   685
Saving account deposits           5600  5880  6068  6205
Money market deposit accounts     1200  1231  1260  1315
Demand and checkable deposits     1010   982   990  1003
Small-denomination time deposits   830   861  1123  1566
Traveler’s checks                    4     4     3     2

In [45]:
M2_df.sum()

2009    10271
2010    10596
2011    11082
2012    11743
dtype: int64

In [53]:
# M2 growth rate
M2_df.sum().pct_change()

2009         NaN
2010    0.031642
2011    0.045866
2012    0.059646
dtype: float64

M2 is broder than M1(1972 / 11743 = 0.17).

## Data analysis problems

1. Go to the St. Louis Federal Reserve FRED database, and find data on currency (CURRSL), traveler’s checks (TVCKSSL), demand deposits (DEMDEPSL), and other checkable deposits (OCDSL). Calculate the M1 money supply, and calculate the percentage change in M1 and in each of the four components of M1 from the most recent month of data available to the same time one year prior. Which component has the highest growth rate? The lowest growth rate? Repeat the calculations using the data from January 2000 to the most recent month of data available, and compare your results.

In [62]:
CURRSL_df   = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=CURRSL&scale=left&cosd=1947-01-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-08&revision_date=2020-03-08&nd=1947-01-01")
TVCKSSL_df  = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=TVCKSSL&scale=left&cosd=1959-01-01&coed=2018-12-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-08&revision_date=2020-03-08&nd=1959-01-01")
DEMDEPSL_df = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=DEMDEPSL&scale=left&cosd=1959-01-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-08&revision_date=2020-03-08&nd=1959-01-01")
OCDSL_df    = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=OCDSL&scale=left&cosd=1963-01-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-08&revision_date=2020-03-08&nd=1963-01-01")

In [89]:
most_recent_month = min(CURRSL_df.tail(1).iat[0, 0], TVCKSSL_df.tail(1).iat[0, 0],
                        DEMDEPSL_df.tail(1).iat[0, 0], OCDSL_df.tail(1).iat[0, 0])
most_recent_month

'2018-12-01'

In [130]:
most_recent_CURRSL   = CURRSL_df.loc[CURRSL_df["DATE"]     == most_recent_month].iat[0, 1]
most_recent_TVCKSSL  = TVCKSSL_df.loc[TVCKSSL_df["DATE"]   == most_recent_month].iat[0, 1]
most_recent_DEMDEPSL = DEMDEPSL_df.loc[DEMDEPSL_df["DATE"] == most_recent_month].iat[0, 1]
most_recent_OCDSL    = OCDSL_df.loc[OCDSL_df["DATE"]       == most_recent_month].iat[0, 1]

most_recent_M1 = most_recent_CURRSL   + \
                 most_recent_TVCKSSL  + \
                 most_recent_DEMDEPSL + \
                 most_recent_OCDSL 
most_recent_M1

3746.5

In [124]:
previous_year = str((int(most_recent_month[:4]) - 1)) + most_recent_month[4:]
previous_year

'2017-12-01'

In [131]:
previous_year_CURRSL   = CURRSL_df.loc[CURRSL_df["DATE"]     == previous_year].iat[0, 1]
previous_year_TVCKSSL  = TVCKSSL_df.loc[TVCKSSL_df["DATE"]   == previous_year].iat[0, 1]
previous_year_DEMDEPSL = DEMDEPSL_df.loc[DEMDEPSL_df["DATE"] == previous_year].iat[0, 1]
previous_year_OCDSL    = OCDSL_df.loc[OCDSL_df["DATE"]       == previous_year].iat[0, 1]

previous_year_M1 = previous_year_CURRSL   + \
                   previous_year_TVCKSSL  + \
                   previous_year_DEMDEPSL + \
                   previous_year_OCDSL
previous_year_M1

3607.3

In [132]:
# M1 grow rate
(most_recent_M1 - previous_year_M1) / previous_year_M1

0.038588417930307935

In [133]:
# currency grow rate
(most_recent_CURRSL - previous_year_CURRSL) / previous_year_CURRSL

0.06544262295081964

In [134]:
# traveler’s checks grow rate
(most_recent_TVCKSSL - previous_year_TVCKSSL) / previous_year_TVCKSSL

-0.10526315789473682

In [135]:
# demand deposits grow rate
(most_recent_DEMDEPSL - previous_year_DEMDEPSL) / previous_year_DEMDEPSL

0.003967720242098245

In [136]:
# other checkable deposits grow rate
(most_recent_OCDSL - previous_year_OCDSL) / previous_year_OCDSL 

0.056791371755982555

From the most recent month to the same time one year prior, currency has the highest growth rate(0.065), traveler’s checks has the lowest(-0.105).

In [137]:
January_2000 = "2000-01-01"

January_2000_CURRSL   = CURRSL_df.loc[CURRSL_df["DATE"]     == January_2000].iat[0, 1]
January_2000_TVCKSSL  = TVCKSSL_df.loc[TVCKSSL_df["DATE"]   == January_2000].iat[0, 1]
January_2000_DEMDEPSL = DEMDEPSL_df.loc[DEMDEPSL_df["DATE"] == January_2000].iat[0, 1]
January_2000_OCDSL    = OCDSL_df.loc[OCDSL_df["DATE"]       == January_2000].iat[0, 1]

January_2000_M1 =  January_2000_CURRSL   + \
                   January_2000_TVCKSSL  + \
                   January_2000_DEMDEPSL + \
                   January_2000_OCDSL
January_2000_M1

1122.1000000000001

In [138]:
# M1 grow rate
(most_recent_M1 - January_2000_M1) / January_2000_M1

2.3388289813742085

In [139]:
# currency grow rate
(most_recent_CURRSL - January_2000_CURRSL) / January_2000_CURRSL

2.095446751762241

In [140]:
# traveler’s checks grow rate
(most_recent_TVCKSSL - January_2000_TVCKSSL) / January_2000_TVCKSSL

-0.7999999999999999

In [141]:
# demand deposits grow rate
(most_recent_DEMDEPSL - January_2000_DEMDEPSL) / January_2000_DEMDEPSL

3.311002021368756

In [142]:
# other checkable deposits grow rate
(most_recent_OCDSL - January_2000_OCDSL) / January_2000_OCDSL

1.587046204620462

From the most recent month to the January 2000, demand deposits has the highest growth rate(3.311), traveler’s checks has the lowest(-0.800).

2. Go to the St. Louis Federal Reserve FRED database, and find data on small-denomination time deposits (STDSL), savings deposits and money market deposit accounts (SAVINGSL), and retail money market funds (RMFSL). Calculate the percentage change of each of these three components of M2 (not included in M1) from the most recent month of data available to the same time one year prior. Which component has the highest growth rate? The lowest growth rate? Repeat the calculations using the data from January 2000 to the most recent month of data available, and compare your results. Use your answers from question 1 to determine which grew faster: the non-M1 components of M2, or the M1 money supply.

In [162]:
STDSL_df    = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=STDSL&scale=left&cosd=1959-01-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-09&revision_date=2020-03-09&nd=1959-01-01")
SAVINGSL_df = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=SAVINGSL&scale=left&cosd=1959-01-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-09&revision_date=2020-03-09&nd=1959-01-01")
RMFSL_df    = pd.read_csv("https://fred.stlouisfed.org/graph/fredgraph.csv?bgcolor=%23e1e9f0&chart_type=line&drp=0&fo=open%20sans&graph_bgcolor=%23ffffff&height=450&mode=fred&recession_bars=on&txtcolor=%23444444&ts=12&tts=12&width=1168&nt=0&thu=0&trc=0&show_legend=yes&show_axis_titles=yes&show_tooltip=yes&id=RMFSL&scale=left&cosd=1973-11-01&coed=2020-01-01&line_color=%234572a7&link_values=false&line_style=solid&mark_type=none&mw=3&lw=2&ost=-99999&oet=99999&mma=0&fml=a&fq=Monthly&fam=avg&fgst=lin&fgsnd=2009-06-01&line_index=1&transformation=lin&vintage_date=2020-03-09&revision_date=2020-03-09&nd=1973-11-01")

most_recent_STDSL    = STDSL_df.loc[STDSL_df["DATE"]       == most_recent_month].iat[0, 1]
most_recent_SAVINGSL = SAVINGSL_df.loc[SAVINGSL_df["DATE"] == most_recent_month].iat[0, 1]
most_recent_RMFSL    = RMFSL_df.loc[RMFSL_df["DATE"]       == most_recent_month].iat[0, 1]

previous_year_STDSL    = STDSL_df.loc[STDSL_df["DATE"]       == previous_year].iat[0, 1]
previous_year_SAVINGSL = SAVINGSL_df.loc[SAVINGSL_df["DATE"] == previous_year].iat[0, 1]
previous_year_RMFSL    = RMFSL_df.loc[RMFSL_df["DATE"]       == previous_year].iat[0, 1]

January_2000_STDSL    = STDSL_df.loc[STDSL_df["DATE"]       == January_2000].iat[0, 1]
January_2000_SAVINGSL = SAVINGSL_df.loc[SAVINGSL_df["DATE"] == January_2000].iat[0, 1]
January_2000_RMFSL    = RMFSL_df.loc[RMFSL_df["DATE"]       == January_2000].iat[0, 1]

In [164]:
# small-denomination time deposits growth rate
(most_recent_STDSL - previous_year_STDSL) / previous_year_STDSL

0.2868872253078965

In [165]:
# savings deposits and money market deposit accounts growth rate
(most_recent_SAVINGSL - previous_year_SAVINGSL) / January_2000_SAVINGSL

0.08647220946256337

In [166]:
# retail money market funds growth rate
(most_recent_RMFSL - previous_year_RMFSL) / previous_year_RMFSL 

0.15302642796248941

From the most recent month to the same time one year prior, small-denomination time deposits has the highest growth rate(0.287), savings deposits and money market deposit accounts has the lowest(0.153).

In [144]:
# small-denomination time deposits growth rate
(most_recent_STDSL - January_2000_STDSL) / January_2000_STDSL

-0.4468548889350218

In [145]:
# savings deposits and money market deposit accounts growth rate
(most_recent_SAVINGSL - January_2000_SAVINGSL) / January_2000_SAVINGSL

4.3174666972898486

In [146]:
# retail money market funds growth rate
(most_recent_RMFSL - January_2000_RMFSL) / January_2000_RMFSL 

-0.020637219406227397

From the most recent month to the January 2000, savings deposits and money market deposit accounts has the highest growth rate(4.317), small-denomination time deposits has the lowest(-0.447).

The non-M1 components of M2 grows faster.